### ISSUE TITLE
Custom wrapper with command chaining (Node-style async trees)

### ISSUE THREAD

**Issue by Ristarg** [aspnet/Microsoft.Data.Sqlite#327]
While working on my project I got destroyed by SQLite errors, so I've made a small wrapper to streamline the database access a bit. It has task-based transaction locking, and also features command execution chains, somewhat reminiscent of Node's callback trains. I figured you guys might appreciate it.

Thanks for all your work. <3

[Link to repo](https://github.com/Ristarg/Microsoft.Data.Sqlite-Database-Controller)

**Comment by bricelam** [comment]
@Ristarg Were there any functional changes we could make in `Microsoft.Data.Sqlite` to make life easier, or was it just about a more streamlined API?

**Comment by bricelam** [comment]
P.S. I find the ADO.NET APIs a bit cumbersome at times too. Hence, [DbConnectionExtensions.cs](https://github.com/aspnet/Microsoft.Data.Sqlite/blob/rel/1.1.0/src/Microsoft.Data.Sqlite/Utilities/DbConnectionExtensions.cs). 😉

**Comment by Ristarg** [comment]
@bricelam I'm not very familiar with the database world, so I can't really comment on the possible improvements, but I learned the "one transaction at a time" rule the hard way (i.e. the 4 hour "why is my code not working" trip), so maybe packing it away into a dispatcher of sorts could be good?Other than that, I've started out making backends on Node, and the verbosity of ADO (and arguably C# in general) is a little rough in comparison. I like when the code flows. So it's mostly an interface thing. I'm pretty happy with my wrapper. At least until something breaks. 👌

**Comment by bricelam** [comment]
Were you getting *locked* errors? #323 should make `BeginTransaction` throw (and block) a lot less.

**Comment by Ristarg** [comment]
I was actually getting locked out on Commit(). Not sure how that works, but when stepping through the execution would just disappear and come back with a locked error _seconds_ later. I tried researching, but all I could find was transactions synchro and usleep timeout issues, so I tried fixing the one I had access too. Although I haven't rewritten the broken modules, so it might still be bork. /shrug
